### ISSUE TITLE
Social plugin fails for missing deps when disabled

### ISSUE THREAD

**Issue by jholdstock** [squidfunk/mkdocs-material#5924]
### Bug description

Using mkdocs 9.2, social plugin is disabled via config, but `mkdocs build` is still failing on `Required dependencies of "social" plugin not found.` This did not happen in 9.1.

### Related links

- https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#+social.enabled
- https://squidfunk.github.io/mkdocs-material/setup/setting-up-social-cards/#+social.cards

### Reproduction

Cannot use info plugin because my build is failing, but I've manually created and attached a zip here: [mkdocs-repro-master.zip](https://github.com/squidfunk/mkdocs-material/files/12474900/mkdocs-repro-master.zip)

### Steps to reproduce

The zip includes a Dockerfile. Running with mkdocs 9.1.21 works, running with 9.2.5 does not.

```no-highlight
docker build . --build-arg mkdocs_version=9.1.21
```

```no-highlight
docker build . --build-arg mkdocs_version=9.2.5
```

### Before submitting

- [X] I have read and followed the [bug reporting guidelines](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/).
- [X] I have attached links to [the documentation](https://squidfunk.github.io/mkdocs-material), and possibly related [issues](https://github.com/squidfunk/mkdocs-material/issues) and [discussions](https://github.com/squidfunk/mkdocs-material/discussions).
- [X] I assure that I have [removed all customizations](https://squidfunk.github.io/mkdocs-material/contributing/reporting-a-bug/#remove-customizations) before submitting this bug report.
- [X] I have attached a __.zip file__ with a [minimal reproduction](https://squidfunk.github.io/mkdocs-material/guides/creating-a-reproduction/).

**Comment by squidfunk** [comment]
Released as part of [9.2.6](https://squidfunk.github.io/mkdocs-material/changelog/#9.2.6).

**Comment by jholdstock** [comment]
Brilliant, amazing response time! Thanks again @squidfunk for everything you do for the project and the wider open source community!
